File: fixtures/scale/objc/large/Families/Feistel/FeistelDescriptor.m
#import "FeistelDescriptor.h"
#import "FeistelCipher.h"
#import "FeistelVectors.h"

@implementation FeistelDescriptor

- (NSString *)family {
    return @"feistel";
}

- (NSString *)suite {
    return @"block";
}

- (id<CLCipher>)cipher {
    return [[FeistelCipher alloc] init];
}

- (NSArray *)vectors {
    return [FeistelVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Feistel/FeistelDescriptor.php
<?php

namespace CipherLab\Families\Feistel;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together FeistelCipher and
// FeistelVectors under the "block" suite.
class FeistelDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'feistel';
    }

    public function suite(): string
    {
        return 'block';
    }

    public function cipher(): Cipher
    {
        return new FeistelCipher();
    }

    public function vectors(): array
    {
        return FeistelVectors::all();
    }
}
